gitops/environment.go

```diff
@@ -0,0 +1,92 @@
+	} else if len(integrationTestScenario.Spec.Environment.Configuration.Env) != 0 {
+		for intEnvVars := range integrationTestScenario.Spec.Environment.Configuration.Env {
+			// if existing environment does not contain EnvVars, copy ones from IntegrationTestScenario
+			envVarFound := false
+			for existingEnvVar := range copiedEnvConfiguration.Env {
+				// envVar names are matching? overwrite existing environment with one from ITS
+				if integrationTestScenario.Spec.Environment.Configuration.Env[intEnvVars].Name == copiedEnvConfiguration.Env[existingEnvVar].Name {
+					copiedEnvConfiguration.Env[existingEnvVar].Value = integrationTestScenario.Spec.Environment.Configuration.Env[intEnvVars].Value
+					envVarFound = true
+				}
+			}
+			if !envVarFound {
+				// in case that EnvVar from IntegrationTestScenario is not matching any EnvVar from existingEnv, add this ITS EnvVar to copied Environment
+				copiedEnvConfiguration.Env = append(copiedEnvConfiguration.Env, applicationapiv1alpha1.EnvVarPair{Name: integrationTestScenario.Spec.Environment.Configuration.Env[intEnvVars].Name, Value: integrationTestScenario.Spec.Environment.Configuration.Env[intEnvVars].Value})
+			}
+		}
+	}
+
+	copyOfEnvironment := applicationapiv1alpha1.Environment{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: existingEnvironment.Name + "-" + integrationTestScenario.Name + "-",
+			Namespace:    namespace,
+		},
+		Spec: applicationapiv1alpha1.EnvironmentSpec{
+			Type:               applicationapiv1alpha1.EnvironmentType_POC,
+			DisplayName:        existingEnvironment.Name + "-" + integrationTestScenario.Name,
+			Tags:               []string{"ephemeral"},
+			DeploymentStrategy: applicationapiv1alpha1.DeploymentStrategy_Manual,
+			Configuration:      copiedEnvConfiguration,
+			UnstableConfigurationFields: &applicationapiv1alpha1.UnstableEnvironmentConfiguration{
+				KubernetesClusterCredentials: applicationapiv1alpha1.KubernetesClusterCredentials{
+					TargetNamespace:          integrationTestScenario.Name + "-" + id.String(),
+					APIURL:                   existingApiURL,
+					ClusterCredentialsSecret: existingClusterCreds,
+				},
+			},
+		},
+	}
+	return &CopiedEnvironment{copyOfEnvironment}
+}
+
+// WithIntegrationLabels adds IntegrationTestScenario name as label to the copied environment.
+func (e *CopiedEnvironment) WithIntegrationLabels(integrationTestScenario *v1alpha1.IntegrationTestScenario) *CopiedEnvironment {
+	if e.ObjectMeta.Labels == nil {
+		e.ObjectMeta.Labels = map[string]string{}
+	}
+	e.ObjectMeta.Labels["test.appstudio.openshift.io/scenario"] = integrationTestScenario.Name
+
+	return e
+
+}
+
+// WithSnapshot adds the name of snapshot as label to the copied environment.
+func (e *CopiedEnvironment) WithSnapshot(snapshot *applicationapiv1alpha1.Snapshot) *CopiedEnvironment {
+
+	if e.ObjectMeta.Labels == nil {
+		e.ObjectMeta.Labels = map[string]string{}
+	}
+	e.ObjectMeta.Labels["test.appstudio.openshift.io/snapshot"] = snapshot.Name
```

Comment:
This will need to be changed to be `appstudio.openshift.io/snapshot`, and also it would be good to define it in a [constant in the snapshot.go](https://github.com/redhat-appstudio/integration-service/blob/main/gitops/snapshot.go#L13)

Reply:
Added, Updated.

Reply:
Thanks!